### ISSUE TITLE
Actions survey

### ISSUE THREAD

**Issue by wendellpiez** [wendellpiez/JATSKit#16]
We have an opportunity to build in more buttons, menu options etc., in some cases by comparing or adapting from other frameworks in oXygen. These bindings are configured by means of Actions at the framework level. Such Actions include support for table manipulation (both HTML and OASIS), but not only these - and of course the potential is open-ended.

On this issue, it would be nice to have a list of nice-to-have Actions.

For example, the Docbook framework has actions for promoting sections (Docbook sect1-sect6 elements); there is no reason that something similar would not work for the JATS/BITS `sec` element.

This may not be a high priority in view of how easy it is to move 'sec' elements around with simple drag-and-drop (in both Author and Outline views). But some people like buttons and menu items - and there are other possible ideas for Actions than these.

I have already added a couple of simple Actions for adding a couple of paragraph-level structures.

Feel free to contribute ideas to this discussion.

**Comment by wendellpiez** [comment]
Possible Actions

- *Insert formula* - inserts `disp-formula` or `inline-formula` depending on what is valid in the document.
- *Insert figure* inserts a `fig` with a stub `caption` and `caption/title`.
- *Insert graphic figure* does the same, except it also includes a `graphic`.

**Comment by juicejuice** [comment]
The problem areas for JATS are of course references/citations, tables and figures. Metadata also plays a big role. Finally the variability of how you are able represent certain content (e.g. funding statements, permissions, floats, footnotes) brings issues. Helper tools for those problem areas would seem to make the most sense if you are looking for ideas :)

**Comment by wendellpiez** [comment]
Agreed. :+1: -- while also keeping in mind that some of these capabilities probably belong in a customization layer not the generic framework. However this is really helpful for focusing attention on the pain points.

BTW I implemented a couple of the Actions listed above.
